wafer/kv/serializers.py

```diff
@@ -0,0 +1,23 @@
+from rest_framework import serializers
+from wafer.kv.models import KeyValue
+
+
+class KeyValueSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = KeyValue
+
+    # There doesn't seem to be a better way of handling the problem
+    # of filtering the groups.
+    # See the DRF meta-issue https://github.com/tomchristie/django-rest-framework/issues/1985
+    # and various related subdisscussions, such as https://github.com/tomchristie/django-rest-framework/issues/2292
+    def __init__(self, *args, **kwargs):
+        # XXX: Breaks introspection tools which call serializers
+        # without a request
```

Comment:
The Group Permission should restrict us to only logged in users already, as someone who isn't logged in doesn't belong to any groups.

The issue with the introspection case is it breaks the assumption that kwargs['context']['request'] is defined, since that is only try if the serializer is called from a proper view. While it's simple enough to return an empty queryset in that case, it's not clear to me whether that's the right thing to do to get sensible results from the introspection tools, which is why I haven't gone that route.